usr/share/laptop-mode-tools/modules/cpuhotplug

```diff
@@ -6,34 +6,67 @@
 
 # Determine number cpu cores on machine
 max_available_cpus() {
-    CPU=/sys/devices/system/cpu/present
-    if [ -f $CPU ]; then
-        max_cpu=$(cat $CPU | cut -d '-' -f2);
-        max_cpu=$(expr $max_cpu + 1);
-        echo $max_cpu;
-    else
-        echo -1;
-    fi
+	CPU=/sys/devices/system/cpu/present
+	if [ -f $CPU ]; then
+		max_cpu=$(cat $CPU | cut -d '-' -f2)
+		max_cpu=$(expr $max_cpu + 1)
+		echo $max_cpu
+	else
+		echo -1
+	fi
+}
+
+# Get a list of cpus to control from a number of cpus that user wants to
+# disable.
+get_cpus_from_number() {
+	counter=0
+	cpu_list=
+
+	for cpu in /sys/devices/system/cpu/cpu[0-9]* ; do
+		if [ $counter -le $1 ]; then
+			counter=$(expr $counter + 1)
+		else
+			break
+		fi
+
+		cpu_list="$cpu_list $cpu"
+	done
 }
 
+# Get a list of cpus from a list of cpu ids. Where each id is of the form
+# cpu<integer>.
+get_cpus_from_ids() {
+	cpu_list=
 
-get_numeric_val() {
-    num_cpu=`max_available_cpus`;
+	for cpu_id in $1 ; do
+		cpu_path="/sys/devices/system/cpu/$cpu_id"
+		if [ -d "$cpu_path" ] ; then
+			cpu_list="$cpu_list $cpu_path"
+		fi
+	done
+}
+
+# Get a list of cpus to control from DISABLE_AVAILABLE_CPU value.
+get_cpus_to_control() {
+	num_cpu=`max_available_cpus`
 
-    case "$1" in
-        "quarter")
-            disable_cpus=$(echo $num_cpu*1/4 | bc);
-            ;;
-        "half")
-            disable_cpus=$(echo $num_cpu*1/2 | bc);
-            ;;
-        "3quarter")
-            disable_cpus=$(echo $num_cpu*3/4 | bc);
-            ;;
-        [0-9]*)
-            disable_cpus=$1;
-            ;;
-    esac
+	case "$1" in
+		"quarter")
+			get_cpus_from_number $(echo $num_cpu*1/4 | bc)
+			;;
+		"half")
+			get_cpus_from_number $(echo $num_cpu*1/2 | bc)
+			;;
+		"3quarter")
+			get_cpus_from_number $(echo $num_cpu*3/4 | bc)
+			;;
+		[0-9]*)
+			get_cpus_from_number $1
+			;;
+		*)
```

Comment:
Don't you mean here to be **cpu*** ? Otherwise, this becomes a catchall. I'd rather want settings be explicit.